src/main/java/com/emc/ecs/servicebroker/service/EcsService.java

```diff
@@ -138,8 +138,10 @@ public CompletableFuture wipeAndDeleteBucket(String bucketName, String namespace
 
             logger.info("Started wipe of bucket '{}' in namespace '{}'", bucketName, namespace);
             BucketWipeResult result = bucketWipeFactory.newBucketWipeResult();
+            bucketWipe.deleteAllMpus(prefix(bucketName), result);
+            bucketWipe.deleteAllVersions(prefix(bucketName), "", result);
```

Comment:
you will likely need to check if versioning is enabled, similar to the way the CLI code does in bucket-wipe - if versioning is enabled, execute `deleteAllVersions`, otherwise `deleteAllObjects`

Reply:
Added condition before calling deleteAllVersions or deleteAllObjects method.